### DISCUSSION TITLE
[Bug ?][openssl] libcrypto-3-x64.dll seems to relie on absolute path for legacy.dll

### DISCUSSION THREAD

**Discussion by cvermot** [microsoft/vcpkg#36403] [Q&A] [answered]
Hi,

I used vcpkg to deploy PoDoFo library in a MSVC project. This library use among other things openssl.
When I execute my application on my dev machine, all is working. When I deploy this app on my target machine, I get an error "_The application was unable to start correctly (0xc0000142)_".

I first thought that it was a DLL problem, so I double checked that all required DLL was on my app side. Everything seemed to be there.

I used Dependency Walker's profiling function which gave me the answer: the openSSL `libcrypto-3-x64.dll` library rely on `legacy.dll`, which seems to be linked with the absolute path of my vcpkg installation on my development computer (DW give me the exact path of legacy.dll on my dev computer) ! If I recreate on my target computer the exact path of legacy.dll on my development computer, my app run !

DW error :
```
DllMain(0x00007FFD7F430000, DLL_PROCESS_ATTACH, 0x000000EEE40FF3A0) in "PODOFO.DLL" called.
LoadLibraryA("C:\vcpkg\packages\openssl_x64-windows\bin\legacy.dll") called from "LIBCRYPTO-3-X64.DLL" at address 0x00007FFD79D5E72E.
LoadLibraryA("C:\vcpkg\packages\openssl_x64-windows\bin\legacy.dll") returned NULL. Error: Le module sp_cifi_ est introuvable (126).
```

I looked over the vcpkg documentation but didn't find any way to configure all dll path references to local directory.

Is it a bug or a vcpkg config problem ?

**Comment by Neumann-A** [comment, marked as answer]
The problem is probably:

```
#ifdef OPENSSLDIR
        return "OPENSSLDIR: \"" OPENSSLDIR "\"";
#else
        return "OPENSSLDIR: N/A";
#endif
    case OPENSSL_ENGINES_DIR:
#ifdef ENGINESDIR
        return "ENGINESDIR: \"" ENGINESDIR "\"";
#else
        return "ENGINESDIR: N/A";
#endif
    case OPENSSL_MODULES_DIR:
#ifdef MODULESDIR
        return "MODULESDIR: \"" MODULESDIR "\"";
#else
        return "MODULESDIR: N/A";
#endif
```

`LIB_CPPFLAGS=-D"L_ENDIAN" -D"OPENSSL_PIC" -D"OPENSSLDIR=\"E:\\vcpkg_folders\\libtorch\\packages\\openssl_x64-windows-release\"" -D"ENGINESDIR=\"E:\\vcpkg_folders\\libtorch\\packages\\openssl_x64-windows-release\\lib\\engines-3\"" -D"MODULESDIR=\"E:\\vcpkg_folders\\libtorch\\packages\\openssl_x64-windows-release\\bin\"" `

Couldn't really find the location where openssl decides to load legacy.dll

**Reply by cvermot** [reply]
Thank you for your answer. From your point of view, this look like a bug ?

**Reply by Neumann-A** [reply]
bug

**Reply by cvermot** [reply]
Thank you. I will transform this conversation to issue !

**Reply by cvermot** [reply]
Issue : https://github.com/microsoft/vcpkg/issues/36482
